### PR TITLE
send Identity id and salesforce account id on enqueued emails

### PR DESCRIPTION
**Pull request by QuarpT**
Required for Braze integration.
If the identity Id is not found, the userId defaults to the sfContactId
sfContactId is set as a custom attribute in Braze
UserId is used when upserting a user to Braze before sending the email

### CODE REVIEW

File: app/services/ExactTargetService.scala [hunk 1]
```diff
@@ -168,32 +168,39 @@ class ExactTargetService(
...
+        numberOfSuspensionsLinedUp = numberOfSuspensionsLinedUp,
+        daysUsed = daysUsed,
+        daysAllowed = daysAllowed
+      )
+    }
 
-    buildDataExtensionRow.flatMap { row =>
-      EitherT(SqsClient.sendDataExtensionToQueue(Config.holidaySuspensionEmailQueue, row).map {
+    def sendToQueue(salesforceContact: Contact, row: HolidaySuspensionBillingScheduleDataExtensionRow) = {
+      SqsClient.sendDataExtensionToQueue(Config.holidaySuspensionEmailQueue, row, salesforceContact.identityId, salesforceContact.salesforceContactId).map {
```
Comment (paulbrown1982) [inline app/services/ExactTargetService.scala]: We might need to double check that the Identity ID has actually been added to the Salesforce contact at this point.

File: app/services/ExactTargetService.scala [hunk 2]
```diff
@@ -261,8 +270,10 @@ object SqsClient extends LazyLogging {
             "SubscriberAttributes" ->  Json.toJsFieldJsValueWrapper(row.fields.toMap)
           )
         ),
-        "DataExtensionName" -> row.forExtension.name
-      ).toString()
+        "DataExtensionName" -> row.forExtension.name,
+        "UserId" -> userId,
+        "SfContactId" -> sfContactId
```
Comment (paulbrown1982) [inline app/services/ExactTargetService.scala]: Having the SfContactId in the data might actually help to speed up the Contact lookup in the workflow service. I think it was searching for one by Identity ID. I'll set a reminder to have a look into that.

Comment (paulbrown1982) [reply app/services/ExactTargetService.scala]: Actually, you could pick up the SF Contact ID in the workflow service, and not need to change subs frontend?

File: app/services/ExactTargetService.scala [hunk 3]
```diff
@@ -168,32 +168,39 @@ class ExactTargetService(
...
+      SqsClient.sendDataExtensionToQueue(Config.holidaySuspensionEmailQueue, row, salesforceContact.identityId, salesforceContact.salesforceContactId).map {
         case Success(sendMsgResult) => \/.right(())
         case Failure(e) => \/.left(s"Details were: ${row.subscriptionName}, ${e.toString}")
-      })
-    }.run.map {
+      }
+    }
+
+    val enqueueResult = for {
+      salesforceContact <- EitherT.right(GetSalesforceContactForSub(subscription)(zuoraService, salesforceService.repo, executionContext))
```
Comment (paulbrown1982) [inline app/services/ExactTargetService.scala]: We need to check that this doesn't require an Identity ID anywhere in the lookup, as holiday suspension emails can go to anyone.

Comment (QuarpT) [reply app/services/ExactTargetService.scala]: You mean for the GetSalesforceContactForSub call? This functionality shouldn't have changed from before

**Comment by QuarpT** [comment]
@paulbrown1982 We were making the `GetSalesforceContactForSub` call in the old code

**Pull request merged**